File: comunidade_app/lib/models/cotacao.dart
class Cotacao {

  final String id;
  final DateTime data;
  final String cooperativa;
  final String produto;
  final double preco;
  final DateTime atualizado;

  Cotacao({
    required this.id,
    required this.data,
    required this.cooperativa,
    required this.produto,
    required this.preco,
    required this.atualizado
  });

  factory Cotacao.fromJson(Map<String, dynamic> json) {
    return Cotacao(
      id: json['id'], 
      data: DateTime.parse(json['data']), 
      cooperativa: json['cooperativa'], 
      produto: json['produto'], 
      preco: json['preco'], 
      atualizado: DateTime.parse(json['atualizado'])
    );
  }

}

File: comunidade_api/src/main/java/com/smaniotto/comunidade_api/model/Cotacao.java
package com.smaniotto.comunidade_api.model;

import java.util.Date;

import com.google.firebase.database.annotations.NotNull;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Cotacao {

    @Id
    @NotNull
    private String id;

    @NotNull
    private Date data;

    @NotNull
    private String cooperativa;

    @NotNull
    private String produto;

    @NotNull
    private Double preco;

    @NotNull
    private Date atualizado;

}
